### PR TITLE
ScreaMachine

### PR DESCRIPTION
**Pull request by MSpekkio**
There's a apparent UI bug with the flipped Villains HP bar disappearing.

### CODE REVIEW

**Comment by wrhyme29** [comment]
The way the advanced start of game effect to reveal cards is programmed, the actual revealing of cards that don't match criteria is hidden. Its not super impactful, as the end result is the same, but it does deviate from the literal effect

**Comment by wrhyme29** [comment]
You might want to add a send message action in TheSetListRevealProcess for when there is no matching keyword so that the player has some additional information about what is happening.

**Comment by wrhyme29** [comment]
The outer foreach loop (looping through cards) is not safe to reactions from other things in the game, as they might modify this list (it would only be if players destroyed one of the bandmates, as band cards are indestructible).

**Comment by wrhyme29** [comment]
Mental Link's implementation doesn't account for reactions that would play extra cards from the Villain Deck. I added a test to illustrate this (TestMentalLink_PlayCardWhenVillainCardIsPlayed) - it passes because i wasn't sure what to assert, but if you look at the output you can see that the activate abilities suppression happens for all cards played in that span, not just the next one.

You might be able to do something where you grab the play index of Mental Link and only suppress for the next card that enters play, but that could get messy with Before timing reactions

**Comment by MSpekkio** [comment]
> 
> 
> Mental Link's implementation doesn't account for reactions that would play extra cards from the Villain Deck. I added a test to illustrate this (TestMentalLink_PlayCardWhenVillainCardIsPlayed) - it passes because i wasn't sure what to assert, but if you look at the output you can see that the activate abilities suppression happens for all cards played in that span, not just the next one.
> 
> You might be able to do something where you grab the play index of Mental Link and only suppress for the next card that enters play, but that could get messy with Before timing reactions

I'll look harder at the cardsource on the incoming activate ability activation, abd maybe match that against the card that was played?

**Comment by MSpekkio** [comment]
> 
> 
> You might want to add a send message action in TheSetListRevealProcess for when there is no matching keyword so that the player has some additional information about what is happening.

It's in the flavor text If no keywords are matched then "Slice is keeping it mellow while...."  I choose flavor over info!

**Comment by wrhyme29** [comment]
> It's in the flavor text If no keywords are matched then "Slice is keeping it mellow while...." I choose flavor over info!

And while I love that, I'm wondering if it makes it hard to follow if you have no idea whats going on with ScreaMachine

**Comment by MSpekkio** [comment]
> 
> 
> The way the advanced start of game effect to reveal cards is programmed, the actual revealing of cards that don't match criteria is hidden. Its not super impactful, as the end result is the same, but it does deviate from the literal effect

Agreed, I don't like long setup's and multireveals just have a bunch of cards flying by.  It's probally safer to follow the text just so the custom card with a 'before start of game reveal trigger' works.

**Review by wrhyme29 — APPROVED** [review]
